### DISCUSSION TITLE
Include parameter name in codegen proto’s Parameter message

### DISCUSSION THREAD

**Discussion by ciarand** [sqlc-dev/sqlc#2845] [Q&A] [answered]
### What do you want to change?

It would be helpful for debugging and better query generation if the codegen/plugin proto’s parameter message included the optional name of the parameter as provided by the author.

I am personally interested in generating better query templates for queries containing both slice params and regular params, which are currently broken (#2452). I haven’t filed an issue for it yet, but reusing the same slice parameter multiple times in the same query also doesn’t work currently (because we use strings.Replace with an N of 1) and would require some annoying counting without using named parameters.

### What database engines need to be changed?

SQLite

**Comment by kyleconroy** [comment, marked as answer]
You can get this information from the column that's passed as part of the parameter. The `is_named_param` field will also be set to `true`.

```sql
-- name: GetAuthor :one
SELECT * FROM authors
WHERE id = sqlc.arg('foo') LIMIT 1;
```

```json
{
  "number": 1,
  "column": {
    "name": "foo",
    "is_named_param": true
  }
}
```
